packages/editor/src/panels/files/loaders.tsx

```diff
@@ -163,35 +162,75 @@ function GeneratingThumbnailsProgress() {
   const { t } = useTranslation()
   const thumbnailJobs = useMutableState(FileThumbnailJobState).jobs
 
-  if (!thumbnailJobs.length) return null
-
-  return (
+  const isLoading = useHookstate(false)
+  const debouncedStatusRef = useRef<ReturnType<typeof setTimeout>>()
+
+  useEffect(() => {
+    clearTimeout(debouncedStatusRef.current)
+  }, [])
+
+  useEffect(() => {
+    if (debouncedStatusRef) {
+      clearTimeout(debouncedStatusRef.current)
+    }
+
+    const isThumbnailsLoading = thumbnailJobs.length > 0
+    if (isThumbnailsLoading) {
+      isLoading.set(true)
+    } else {
+      debouncedStatusRef.current = setTimeout(() => {
+        isLoading.set(false)
+      }, 1000)
```

Comment:
Why was a return statement not used to clear the timeout? Was it not working since the timeout isn't always set?